lib/fluent/plugin/in_tail.rb

```diff
@@ -655,18 +692,21 @@ def on_rotate(stat)
             watcher_needs_update = true
           end
 
-          log_msg = "detected rotation of #{@path}"
-          log_msg << "; waiting #{@rotate_wait} seconds" if watcher_needs_update # wait rotate_time if previous file exists
-          @log.info log_msg
-
           if watcher_needs_update
             @update_watcher.call(@path, swap_state(@pe))
           else
-            @io_handler = IOHandler.new(self, &method(:wrap_receive_lines))
+            @log.info "detected rotation of #{@path}"
```

Comment:
Is this line needed?

Reply:
(L658)[https://github.com/fluent/fluentd/pull/2832/files#diff-1da710c9dcc8d0fc57996df7a9d39695L658] and (L660)[https://github.com/fluent/fluentd/pull/2832/files#diff-1da710c9dcc8d0fc57996df7a9d39695L660] is invoked even if `watcher_needs_update` is `false`, right?

Reply:
Ah, I see.